packages/flutter_tools/test/src/testbed.dart

```diff
@@ -7,15 +7,32 @@ import 'dart:async';
 import 'package:file/memory.dart';
 import 'package:flutter_tools/src/base/context.dart';
 import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/io.dart';
 import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/os.dart';
+import 'package:flutter_tools/src/base/terminal.dart';
 import 'package:flutter_tools/src/cache.dart';
 import 'package:flutter_tools/src/context_runner.dart';
+import 'package:flutter_tools/src/device.dart';
+import 'package:flutter_tools/src/doctor.dart';
+import 'package:flutter_tools/src/usage.dart';
+import 'package:flutter_tools/src/version.dart';
+
+import 'context.dart';
 
 export 'package:flutter_tools/src/base/context.dart' show Generator;
 
 final Map<Type, Generator> _testbedDefaults = <Type, Generator>{
   FileSystem: () => MemoryFileSystem(),
   Logger: () => BufferLogger(),
+  DeviceManager: () => MockDeviceManager(),
```

Comment:
This list needs some comments describing how its membership is decided. What is the procedure for deciding that a mock should be in this defaults list?

Reply:
Added a standard criteria and removed those that did not fit. Also documented the reason for each current defaults inclusion